### ISSUE TITLE
APL hyper-parameter tunning 

### ISSUE THREAD

**Issue by mak-rayate** [SAP-samples/hana-ml-samples#28]
Hello , 
Do we need hyperparameter tuning while using SAP APL models ? 
as per document there are default values for learning_rate,max depth ,max_iterations,early stopping . After using this default values my model is giving me 81% ACC on training dataset. But I need to further improve the model ACC. I got nowhere about hyper-parameter tuning about APL models. 
In APL, how can I tune above parameters , do we have any options to activate correct auto-selection for above hyperparameters or model do it by themselves at backend and no need to do any hyperparameter tuning for APL models ?

my code : 
`apl_model = GradientBoostingClassifier()
    apl_model.set_params(variable_auto_selection = True,variable_selection_max_nb_of_final_variables = '6')
    apl_model.fit(hdf_train, label=col_target, key=col_key,features=col_predictors)`

**Comment by MarcDaniau** [comment]
Hello,
For gradient boosting hyperparameters, APL applies a set of default values that works well on average for most ML cases. One can change these default values manually if needed. 
Example:
`apl_model.set_params(other_train_apl_aliases={'APL/LearningRate':'0.025'})`

APL does not provide grid search capability to tune the hyperparameters. An option would be to use Optuna for example.

**Comment by mak-rayate** [comment]
Hey @MarcDaniau , 
Thank you for your response .  I will try Optuna to find hyperparameters and will let you know if any issue occurred. For now closing the incident.
